cypress/integration/EditAndBranching/TagAndFrog.spec.js

```diff
@@ -8,34 +8,46 @@ import * as modelPage from '../../support/components/ModelPage'
 import * as scorerModal from '../../support/components/ScorerModal'
 import * as train from '../../support/Train'
 import * as editDialogModal from '../../support/components/EditDialogModal'
+import * as helpers from '../../support/Helpers'
 
-describe('EditAndBranching', () => {
-  it('Tag And Frog', () => {
-    const textEntityPairs = [{text: 'Tag', entity: 'multi'}, {text: 'Frog', entity: 'multi'}]
+describe('Tag and Frog - Edit And Branching', () => {
+  const textEntityPairs = [{text: 'Tag', entity: 'multi'}, {text: 'Frog', entity: 'multi'}]
 
+  afterEach(helpers.SkipRemainingTestsOfSuiteIfFailed)
+  
+  context('Setup', () => {
+    it('Should import a model and wait for training to complete', () => {
     models.ImportModel('z-tagAndFrog2', 'z-tagAndFrog2.cl')
     modelPage.NavigateToTrainDialogs()
     cy.WaitForTrainingStatusCompleted()
+    })
+  })
 
-    train.EditTraining('This is Tag.', 'This is Tag.', 'Hi')
-    editDialogModal.SelectChatTurnExactMatch('This is Tag.')
+  context('Edit Train Dialog', () => {
+    it('Should remove two entity labels from alternative input', () => {
+      train.EditTraining('This is Tag.', 'This is Tag.', 'Hi')
+      editDialogModal.SelectChatTurnExactMatch('This is Tag.')
 
-    editDialogModal.VerifyEntityLabelWithinSpecificInput([textEntityPairs[0]], 0)
-    editDialogModal.VerifyEntityLabelWithinSpecificInput(textEntityPairs, 1)
-    editDialogModal.VerifyEntityLabelWithinSpecificInput(textEntityPairs, 2)
+      editDialogModal.VerifyEntityLabelWithinSpecificInput([textEntityPairs[0]], 0)
```

Comment:
Why does the first call use `[textEntityPairs[0]]` and not the rest?  I see some weird logic around indices of slate editor but looks unecessary.

Reply:
In this case there are three user inputs being manipulated:

1. The main user utterance with 'Tag' being labeled as 'multi' entity.
2. The 1st alternative input, with both entity pairs in that array being labeled.
3. the 2nd alternative input, with both entity pairs in that array being labeled.

As far as specifying the indices of the "slate editor"...Either I have to add logic to the Verify function that finds the user utterance based on a string, or I can specify it as an index. In this case it seemed better to use an index, but I think the spec code is easier to read and understand when it is a string. I would be happy to change it to a string if you also think that is better. Or do you have another suggestion on how to handle such a function?